backend/src/main/java/com/huellapositiva/application/controller/ProposalApiController.java

```diff
@@ -0,0 +1,124 @@
+package com.huellapositiva.application.controller;
+
+import com.huellapositiva.application.dto.ProposalRequestDto;
+import com.huellapositiva.application.dto.ProposalResponseDto;
+import com.huellapositiva.application.exception.ProposalNotPublished;
+import com.huellapositiva.domain.actions.FetchProposalAction;
+import com.huellapositiva.domain.actions.JoinProposalAction;
+import com.huellapositiva.domain.actions.RegisterProposalAction;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.annotation.security.RolesAllowed;
+import javax.persistence.EntityNotFoundException;
+
+@RestController
+@AllArgsConstructor
+@Tag(name = "Proposal Service", description = "The proposals API")
+@RequestMapping("/api/v1/proposals")
+public class ProposalApiController {
+
+    private final RegisterProposalAction registerProposalAction;
+
+    private final FetchProposalAction fetchProposalAction;
+
+    private final JoinProposalAction joinProposalAction;
+
+    @Operation(
+            summary = "Register a new proposal",
+            description = "Register a new proposal and link it to the logged employee",
+            tags = "proposals"
+    )
+    @ApiResponses(
+            value = {
+                    @ApiResponse(
+                            responseCode = "201",
+                            description = "Ok, proposal register successful"
+                    )
+            }
+    )
+    @PostMapping
+    @RolesAllowed("ORGANIZATION_MEMBER")
+    @ResponseBody
+    @ResponseStatus(HttpStatus.CREATED)
+    public void createProposal(@RequestBody ProposalRequestDto dto) {
+        String employeeEmail = (String) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
```

Comment:
You can inject the Authentication object as a parameter to the method instead of using static methods of the SecurityContext which are not very maintainable.